### PR TITLE
feat: add declarations to git

### LINKED ISSUE #74: Add declarations to git
### Expected Behavior / Situation

dist folder presented in git, so we can see changes in types on each PR

### Actual Behavior / Situation

dist folder is not presented in git

### Modification Proposal

I suggest to follow same approach as core repo, I agree with @sokra idea https://github.com/webpack/webpack/pull/9939#issuecomment-551425142

So we can check declarations changes as we check snapshot changes

### PR DESCRIPTION
**Pull request by vankop**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case

closes #74 

### Breaking Changes

 no

### Additional Info

no

### CODE REVIEW

**Comment by vankop** [comment]
> Let's keep `dist` ignored and put types in other directory, for example `types/schema-utils/`?

yep, what do you think about cleaning up this directory before `build:types`? I think it good practice to prevent delivering useless code (for example when validate will be renamed, declaration  for `validate.js` will be still presented).

**Comment by evilebottnawi** [comment]
@vankop yep, we can create new npm task like `clean:types`

File: package.json [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-  "types": "dist/index.d.ts",
+  "types": "declarations/index.d.ts",
   "engines": {
     "node": ">= 8.9.0"
   },
   "scripts": {
     "start": "npm run build -- -w",
     "prebuild": "npm-run-all --serial clean build:types",
-    "build:types": "tsc --declaration --emitDeclarationOnly --outDir dist",
+    "build:types": "rimraf declarations && tsc --declaration --emitDeclarationOnly --outDir declarations && prettier \"declarations/**/*.ts\" --write",
```
Comment (sokra) [inline package.json]: You probably want to check as part of the CI process if the declarations are up-to-date. So contributors do not forget to build the declarations

**Comment by vankop** [comment]
> Need add `declarations` here

@evilebottnawi fixed

File: package.json (hunk 1 above)
Comment (vankop) [reply package.json]: You are correct, but I can not figure it out how to do it better, looks like for webpack core repo it also can be a problem (if declarations and JSON schemas will be created from code)

Maybe add some *special* step in CI where we bash script aka:
```
npm run build
if [ -n "$(git status --porcelain)" ]; then
  echo "Missing declarations in git";
  exit 1;
else
  echo "All declarations are valid";
fi
```

So if this step fails contributor can get some info about it from `webpack-bot` +
maybe someone wants to implement autocommit for `webpack bot` =)

Comment (vankop) [reply package.json]: /cc @evilebottnawi what do you think about it?

Comment (evilebottnawi) [reply package.json]: @vankop let's solve this in other PR, can you open issue?

**Pull request merged**